File: lv-vhdl-frontend/src/main/java/vhdl2vhdl.java
import java.io.FileInputStream;

import stupaq.ExceptionPrinter;
import stupaq.vhdl93.VHDL93Parser;
import stupaq.vhdl93.ast.SimpleNode;
import stupaq.vhdl93.transformers.FlattenNestedListsVisitor;
import stupaq.vhdl93.visitor.TreeDumper;
import stupaq.vhdl93.formatting.VHDLTreeFormatter;

public class vhdl2vhdl {
  public static void main(String args[]) {
    try {
      if (args.length == 1) {
        FileInputStream file = new FileInputStream(args[0]);
        VHDL93Parser parser = new VHDL93Parser(file);
        SimpleNode root = parser.design_file();
        root.accept(new FlattenNestedListsVisitor());
        root.accept(new TreeDumper(System.err));
        System.err.println();
        root.accept(new VHDLTreeFormatter());
        root.accept(new TreeDumper(System.out));
        System.out.println();
      } else {
        System.err.println("usage: filename");
      }
    } catch (Exception e) {
      ExceptionPrinter.print(e, System.err);
    }
  }
}

File: lv-vhdl-frontend/src/main/scala/vhdl2vhdl.scala
import java.io.FileInputStream

import stupaq.translation.ExceptionPrinter
import stupaq.translation.semantic.FlattenNestedListsVisitor
import stupaq.vhdl93.VHDL93ParserTotal
import stupaq.vhdl93.formatting.VHDLTreeFormatter
import stupaq.vhdl93.visitor.TreeDumper

object vhdl2vhdl {

  def main(args: Array[String]) {
    try {
      if (args.length == 1) {
        val file = new FileInputStream(args(0))
        val parser = new VHDL93ParserTotal(file)
        val root = parser.design_file()
        root.accept(new TreeDumper(System.out))
        println()
        println(new String(new Array[Char](40)).replace('\u0000', '-'))
        root.accept(new FlattenNestedListsVisitor())
        root.accept(new VHDLTreeFormatter())
        root.accept(new TreeDumper(System.out))
        println()
      } else {
        println("usage: filename")
      }
    } catch {
      case e: Exception => ExceptionPrinter.print(e, System.err)
    }
  }
}
